fe/src/main/java/org/apache/doris/analysis/AdminSetReplicaStatusStmt.java

```diff
@@ -0,0 +1,119 @@
+
+import java.util.List;
+import java.util.Map;
+
+/*
+ *  admin set replicas status properties ("key" = "val", ..);
+ *  Required:
+ *      "tablet_id" = "1,2,3,4,5,...",
+ *      "backend_id" = "10001"
+ *      "status" = "bad"
+ */
+public class AdminSetReplicaStatusStmt extends DdlStmt {
+
+    public static final String TABLET_ID = "tablet_id";
+    public static final String BACKEND_ID = "backend_id";
+    public static final String STATUS = "status";
+    public static final String STATUS_BAD = "bad";
+
+    private Map<String, String> properties;
+    private List<Long> tabletIds = Lists.newArrayList();
+    private long backendId = -1;
+    private String status = null;
+
+    public AdminSetReplicaStatusStmt(Map<String, String> properties) {
+        this.properties = properties;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
+        super.analyze(analyzer);
+
+        // check auth
+        if (!Catalog.getCurrentCatalog().getAuth().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+        }
+
+        checkProperties();
+    }
+
+    private void checkProperties() throws AnalysisException {
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            String key = entry.getKey();
+            String val = entry.getValue();
+
+            if (key.equalsIgnoreCase(TABLET_ID)) {
+                try {
+                    String[] ids = val.replace(" ", "").split(",");
+                    for (String idStr : ids) {
+                        tabletIds.add(Long.valueOf(idStr));
+                    }
+                } catch (NumberFormatException e) {
+                    throw new AnalysisException("Invalid tablet id format: " + val);
+                }
+            } else if (key.equalsIgnoreCase(BACKEND_ID)) {
+                try {
+                    backendId = Long.valueOf(val);
+                } catch (NumberFormatException e) {
+                    throw new AnalysisException("Invalid backend id format: " + val);
+                }
+            } else if (key.equalsIgnoreCase(STATUS)) {
+                status = val.toLowerCase();
+                if (!status.equals(STATUS_BAD)) {
+                    throw new AnalysisException("Do not support setting replica status as " + val);
+                }
+            } else {
+                throw new AnalysisException("Unknown property: " + key);
+            }
+        }
+
+        if (tabletIds.isEmpty() || backendId == -1 || status == null) {
+            throw new AnalysisException("Should add following properties: TABLET_ID, BACKEND_ID and STATUS");
+        }
+    }
+
+    public List<Long> getTabletIds() {
+        return tabletIds;
+    }
+
+    public long getBackendId() {
+        return backendId;
+    }
+
+    public String getStatus() {
```

Comment:
I think it's better to introduce an enum type here to avoid string compare

Reply:
OK